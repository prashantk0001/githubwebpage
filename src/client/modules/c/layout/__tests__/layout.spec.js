/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import cElement from 'c/layout';

function createComponent(props = {}) {
    const element = createElement('c-layout', { is: cElement });
    Object.assign(element, props);
    document.body.appendChild(element);
    return element;
}

describe('c-layout-item', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('default', () => {
        const element = createComponent();

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('align', () => {
        const element = createComponent({
            horizontalAlign: 'center',
            verticalAlign: 'end'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('pullToBoundary', () => {
        const element = createComponent({
            horizontalAlign: 'center',
            verticalAlign: 'end',
            pullToBoundary: 'small'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('multiRows', () => {
        const element = createComponent({
            horizontalAlign: 'center',
            verticalAlign: 'end',
            multiRows: true
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
