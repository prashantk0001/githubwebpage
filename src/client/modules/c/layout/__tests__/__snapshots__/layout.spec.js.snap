// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-layout-item align 1`] = `
<c-layout
  class="slds-grid slds-grid_align-center slds-grid_vertical-align-end"
>
  #shadow-root(open)
    <slot
      class="slds-slot"
    />
</c-layout>
`;

exports[`c-layout-item default 1`] = `
<c-layout
  class="slds-grid"
>
  #shadow-root(open)
    <slot
      class="slds-slot"
    />
</c-layout>
`;

exports[`c-layout-item multiRows 1`] = `
<c-layout
  class="slds-grid slds-grid_align-center slds-grid_vertical-align-end"
>
  #shadow-root(open)
    <slot
      class="slds-slot"
    />
</c-layout>
`;

exports[`c-layout-item pullToBoundary 1`] = `
<c-layout
  class="slds-grid slds-grid_align-center slds-grid_vertical-align-end slds-grid_pull-padded"
>
  #shadow-root(open)
    <slot
      class="slds-slot"
    />
</c-layout>
`;
